sms_sponsorship/webapp/src/App.js

```diff
@@ -99,81 +95,80 @@ class Main extends React.Component {
                 <div>
                     {topAppBar}
                     <div>
-                        <Message text={t("error_noRequestID")}/>
+                        <Error appContext={this} text="Error : no SMS request ID. Please send an other SMS."/>
                     </div>
                 </div>
             )
         }
 
-        if (!child.has_a_child && !this.state.success) {
+        if (!child.has_a_child) {
             clearTimeout(window.getChildTimeout);
-            window.getChildTimeout = setTimeout(() => {
+            if(this.count_try > 20){
+                return (
+                            <div>
+                                <Error appContext={this} text="Error : No child found. Try again later"/>
+                            </div>
+                        )
+            }else{
+                window.getChildTimeout = setTimeout(() => {
                 this.getChild();
-            }, 1000);
+                }, 1000);
+                ++this.count_try;
+            }
+        }
+        if (this.state.success) {
+            return (
+                <div>success !</div>
```

Comment:
This is again a very bad revert that cancels a lot of changes we made. We have a better success message than this.